src/entrypoints/options/providers/base-url-config.tsx

```diff
@@ -0,0 +1,62 @@
+import type { APIProviderNames } from '@/types/config/provider'
+import { useAtom } from 'jotai'
+import { useCallback } from 'react'
+import ProviderIcon from '@/components/provider-icon'
+import { Input } from '@/components/ui/input'
+import { configFields } from '@/utils/atoms/config'
+import { API_PROVIDER_ITEMS } from '@/utils/constants/config'
+
+export default function BaseURLConfig() {
+  const [providersConfig, setProvidersConfig] = useAtom(configFields.providersConfig)
+
+  const handleBaseUrlChange = useCallback((provider: APIProviderNames, newBaseUrl: string) => {
+    setProvidersConfig({
+      [provider]: {
+        ...providersConfig[provider],
+        baseURL: newBaseUrl,
+      },
+    })
+  }, [setProvidersConfig, providersConfig])
+
+  // get placeholder for each provider
+  const getPlaceholder = (provider: string) => {
```

Comment:
is there any other way to not code this in the component level? like move the logic to constants folder, or don't need switch is even better